renderdoc/driver/vulkan/vk_core.h

```diff
@@ -735,59 +746,44 @@ class WrappedVulkan : public IFrameCapturer
   };
   rdcarray<ActionUse> m_ActionUses;
 
-  enum PartialReplayIndex
-  {
-    Primary,
-    Secondary,
-    ePartialNum
-  };
-
-  struct Submission
-  {
-    Submission(uint32_t eid) : baseEvent(eid), rebased(false) {}
-    uint32_t baseEvent = 0;
-    bool rebased = false;
-  };
-
-  // by definition, when replaying we must have N completely submitted command buffers, and at most
-  // two partially-submitted command buffers. One primary, that we're part-way through, and then
-  // if we're part-way through a vkCmdExecuteCommandBuffers inside that primary then there's one
-  // secondary.
+  // during active replay, command buffers may be partially-submitted if the selected event occurs
+  // within the range of the command buffer. If secondary command buffers are used and the selected
+  // event occurs within the range of a secondary, both the secondary and the command buffer that
+  // executed it are considered partially-submitted. If VK_EXT_nested_command_buffer is enabled,
+  // there may be up to N partially-submitted command buffers, where N is the maximum command
+  // nesting depth used in the capture.
   struct PartialReplayData
   {
     PartialReplayData() { Reset(); }
     void Reset()
     {
-      partialParent = ResourceId();
-      baseEvent = 0;
-      renderPassActive = false;
+      isPartialPrimary = false;
+      currentPartial = ResourceId();
+      partialSubmits.clear();
     }
 
-    // this records where in the frame a command buffer was submitted, so that we know if our replay
-    // range ends in one of these ranges we need to construct a partial command buffer for future
+    // this tracks the command buffers that are actually partial during replay. These are used to
+    // track whether a render pass is active or not during partial replay. For the most part, we
+    // only care about the partial command buffer we are currently replaying, but there is a case
+    // where we need to save the state of the partial stack and restore it (see ReplayLog())
+    std::map<ResourceId, Submission> partialSubmits;
+    bool isPartialPrimary;
+    ResourceId currentPartial;
```

Comment:
I don't think that storing only one command buffer here is sufficient.

This information is used for only a couple of things in the end:

1. For knowing if this is the (single) primary partial command buffer. For updating RP state via `WrappedVulkan::ShouldUpdateRenderState()` and for closing any pending partially-replayed things like render passes, XFB, conditional rendering.
2. For knowing the _innermost_ partial command buffer (whether it be primary or secondary) to know which one to pull render state from via `WrappedVulkan::GetPartialCommandBuffer()`.
3. For knowing if this is _any_ partial command buffer to close a pending dynamic rendering instance which could have been begun in any secondary command buffer.

So I think that means we need to have an equivalent knowledge to previously of knowing the full list of N partial command buffers all the way down to the innermost.